[Bug]: SFML and GLFW doesn't work

### Problem description

Two newly added packages glfw and sfml by @xtkoba are not working properly, at least on my device. Here's the errors for both 
```
➜  SFML g++ -pedantic-errors -o main.exe main.cpp -lsfml-graphics -lsfml-window -lsfml-system -lsfml-audio -lsfml-network
ld.lld: error: /data/data/com.termux/files/usr/lib/libsfml-system.so: undefined reference to sf::priv::ResourceStream::ResourceStream(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&) [--no-allow-shlib-undefined]
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
GLFW:-
➜  GLFW ./main.exe
CANNOT LINK EXECUTABLE "./main.exe": cannot locate symbol "_glfwInitJoysticksLinux" referenced by "/data/data/com.termux/files/usr/lib/libglfw.so"...
```
Maybe I am doing something wrong so I will be very grateful if someone provides an working example.
I was compiling raylib which uses glfw and faced the same error that _glfwInitJoysticksLinux is missing.

### What steps will reproduce the bug?

Install SFML and GLFW and try to run the basic example code.

### What is the expected behavior?

_No response_

### System information

termux-info:

```
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PID=11997
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://packages-cf.termux.org/apt/termux-main/ stable main
# sources.list.d/termux-desktop-xfce.list
deb [trusted=yes arch=all] https://Yisus7u7.github.io/termux-desktop-xfce/packages desktop-xfce extras
# x11-repo (sources.list.d/x11.list)
deb https://packages.termux.org/apt/termux-x11 x11 main
Updatable packages:
curl/stable 7.83.0 arm [upgradable from: 7.82.0-1]
libcurl/stable 7.83.0 arm [upgradable from: 7.82.0-1]
lsof/stable 4.95.0 arm [upgradable from: 4.94.0-1]
nnn/stable 4.5 arm [upgradable from: 4.4-1]
termux-am-socket/stable 1.3.0 arm [upgradable from: 1.02]
Android version:
10
Kernel build information:
Linux localhost 4.4.177-19000394 #1 SMP PREEMPT Mon Jul 27 19:29:08 KST 2020 armv8l Android
Device manufacturer:
samsung
Device model:
```

--- a/x11-packages/glfw/build.sh
+++ b/x11-packages/glfw/build.sh
@@ -3,6 +3,7 @@ TERMUX_PKG_DESCRIPTION="An Open Source, multi-platform library for OpenGL, OpenG
 TERMUX_PKG_LICENSE="ZLIB"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=3.3.7
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://github.com/glfw/glfw/archive/refs/tags/${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=fd21a5f65bcc0fc3c76e0f8865776e852de09ef6fbc3620e09ce96d2b2807e04
 TERMUX_PKG_DEPENDS="libx11"
@@ -13,3 +14,7 @@ TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 -DGLFW_BUILD_TESTS=OFF
 -DGLFW_BUILD_DOCS=OFF
 "
+
+termux_step_pre_configure() {
+	LDFLAGS+=" -lm"
+}

--- a/x11-packages/glfw/src-wl_init.c.patch
+++ b/x11-packages/glfw/src-wl_init.c.patch
@@ -1,0 +1,20 @@
+--- a/src/wl_init.c
++++ b/src/wl_init.c
+@@ -1150,7 +1150,7 @@
+     // Sync so we got all initial output events
+     wl_display_roundtrip(_glfw.wl.display);
+ 
+-#ifdef __linux__
++#if defined __linux__ && !defined __ANDROID__
+     if (!_glfwInitJoysticksLinux())
+         return GLFW_FALSE;
+ #endif
+@@ -1211,7 +1211,7 @@
+ 
+ void _glfwPlatformTerminate(void)
+ {
+-#ifdef __linux__
++#if defined __linux__ && !defined __ANDROID__
+     _glfwTerminateJoysticksLinux();
+ #endif
+     _glfwTerminateEGL();

--- a/x11-packages/glfw/src-wl_platform.h.patch
+++ b/x11-packages/glfw/src-wl_platform.h.patch
@@ -1,0 +1,11 @@
+--- a/src/wl_platform.h
++++ b/src/wl_platform.h
+@@ -45,7 +45,7 @@
+ 
+ #include "posix_thread.h"
+ #include "posix_time.h"
+-#ifdef __linux__
++#if defined __linux__ && !defined __ANDROID__
+ #include "linux_joystick.h"
+ #else
+ #include "null_joystick.h"

--- a/x11-packages/glfw/src-x11_init.c.patch
+++ b/x11-packages/glfw/src-x11_init.c.patch
@@ -63,3 +63,21 @@
      _glfw.x11.xrender.handle = _glfw_dlopen("libXrender.so");
  #else
      _glfw.x11.xrender.handle = _glfw_dlopen("libXrender.so.1");
+@@ -1147,7 +1147,7 @@
+         }
+     }
+ 
+-#if defined(__linux__)
++#if defined(__linux__) && !defined(__ANDROID__)
+     if (!_glfwInitJoysticksLinux())
+         return GLFW_FALSE;
+ #endif
+@@ -1240,7 +1240,7 @@
+     _glfwTerminateEGL();
+     _glfwTerminateGLX();
+ 
+-#if defined(__linux__)
++#if defined(__linux__) && !defined(__ANDROID__)
+     _glfwTerminateJoysticksLinux();
+ #endif
+ 

--- a/x11-packages/glfw/src-x11_window.c.patch
+++ b/x11-packages/glfw/src-x11_window.c.patch
@@ -1,0 +1,11 @@
+--- a/src/x11_window.c
++++ b/src/x11_window.c
+@@ -2830,7 +2830,7 @@
+ {
+     drainEmptyEvents();
+ 
+-#if defined(__linux__)
++#if defined(__linux__) && !defined(__ANDROID__)
+     _glfwDetectJoystickConnectionLinux();
+ #endif
+     XPending(_glfw.x11.display);

--- a/x11-packages/sfml/build.sh
+++ b/x11-packages/sfml/build.sh
@@ -3,17 +3,16 @@ TERMUX_PKG_DESCRIPTION="A simple, fast, cross-platform and object-oriented multi
 TERMUX_PKG_LICENSE="ZLIB"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=2.5.1
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://github.com/SFML/SFML/archive/refs/tags/${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=438c91a917cc8aa19e82c6f59f8714da353c488584a007d401efac8368e1c785
 TERMUX_PKG_DEPENDS="freetype, libc++, libflac, libogg, libvorbis, libx11, libxrandr, mesa, openal-soft"
 
 termux_step_post_get_source() {
-	cp src/SFML/Window/Android/JoystickImpl.cpp src/SFML/Window/Unix/
+	cp src/SFML/Window/Android/JoystickImpl.{cpp,hpp} src/SFML/Window/Unix/
 }
 
 termux_step_pre_configure() {
-	CPPFLAGS+=" -DSFML_SYSTEM_LINUX"
-
 	_NEED_DUMMY_LIBPTHREAD_A=
 	_LIBPTHREAD_A=$TERMUX_PREFIX/lib/libpthread.a
 	if [ ! -e $_LIBPTHREAD_A ]; then

--- a/x11-packages/sfml/include-SFML-Config.hpp.patch
+++ b/x11-packages/sfml/include-SFML-Config.hpp.patch
@@ -1,0 +1,16 @@
+--- a/include/SFML/Config.hpp
++++ b/include/SFML/Config.hpp
+@@ -71,12 +71,7 @@
+ #elif defined(__unix__)
+ 
+     // UNIX system, see which one it is
+-    #if defined(__ANDROID__)
+-
+-        // Android
+-        #define SFML_SYSTEM_ANDROID
+-
+-    #elif defined(__linux__)
++    #if defined(__linux__)
+ 
+          // Linux
+         #define SFML_SYSTEM_LINUX

--- a/x11-packages/sfml/src-SFML-System-Unix-ThreadImpl.cpp.patch
+++ b/x11-packages/sfml/src-SFML-System-Unix-ThreadImpl.cpp.patch
@@ -1,0 +1,20 @@
+--- a/src/SFML/System/Unix/ThreadImpl.cpp
++++ b/src/SFML/System/Unix/ThreadImpl.cpp
+@@ -62,7 +62,7 @@
+ {
+     if (m_isActive)
+     {
+-        #ifndef SFML_SYSTEM_ANDROID
++        #ifndef __ANDROID__
+             pthread_cancel(m_thread);
+         #else
+             // See https://stackoverflow.com/questions/4610086/pthread-cancel-al
+@@ -78,7 +78,7 @@
+     // The Thread instance is stored in the user data
+     Thread* owner = static_cast<Thread*>(userData);
+ 
+-    #ifndef SFML_SYSTEM_ANDROID
++    #ifndef __ANDROID__
+         // Tell the thread to handle cancel requests immediately
+         pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
+     #endif

